soccer/src/soccer/planning/planner_node.cpp

```diff
@@ -307,63 +317,75 @@ PlanRequest PlannerForRobot::make_request(const RobotIntent& intent) {
+                                             robot_id_, request.motion_command.name));
+    }
 
-        // If it fails, or if the planner was not used, the trajectory will
-        // still be empty. Reset the planner.
-        if (trajectory.empty()) {
-            current_planner_ = nullptr;
-            planner->reset();
-        } else {
-            if (!trajectory.angles_valid()) {
-                throw std::runtime_error("Trajectory returned from " + planner->name() +
-                                         " has no angle profile!");
-            }
+    // get Trajectory from the planner requested in MotionCommand
+    current_planner_ = planners_[request.motion_command.name];
+    Trajectory trajectory = current_planner_->plan(request);
 
-            if (!trajectory.time_created().has_value()) {
-                throw std::runtime_error("Trajectory returned from " + planner->name() +
-                                         " has no timestamp!");
-            }
-        }
+    if (trajectory.empty()) {
+        // empty Trajectory means current_planner_ has failed
+        // if current_planner_ fails, reset it before throwing exception
+        current_planner_->reset();
+        throw std::runtime_error(fmt::format("PathPlanner <{}> failed to create valid Trajectory!",
+                                             current_planner_->name()));
     }
 
-    if (trajectory.empty()) {
-        SPDLOG_ERROR("No valid planner! Did you forget to specify a default planner?");
-        trajectory = Trajectory{{request.start}};
-        trajectory.set_debug_text("Error: No Valid Planners");
-    } else {
-        // draw robot's desired path
-        std::vector<rj_geometry::Point> path;
-        std::transform(trajectory.instants().begin(), trajectory.instants().end(),
-                       std::back_inserter(path),
-                       [](const auto& instant) { return instant.position(); });
-        debug_draw_.draw_path(path);
-
-        // draw robot's desired endpoint
-        debug_draw_.draw_circle(rj_geometry::Circle(path.back(), kRobotRadius), Qt::black);
+    if (!trajectory.angles_valid()) {
+        throw std::runtime_error(fmt::format("Trajectory returned from <{}> has no angle profile!",
+                                             current_planner_->name()));
     }
-    debug_draw_.draw_shapes(shared_state_->global_obstacles(), QColor(255, 0, 0, 30));
-    debug_draw_.draw_shapes(request.virtual_obstacles, QColor(255, 0, 0, 30));
 
+    if (!trajectory.time_created().has_value()) {
+        throw std::runtime_error(fmt::format("Trajectory returned from <{}> has no timestamp!",
+                                             current_planner_->name()));
+    }
+
+    return trajectory;
+}
+
+Trajectory PlannerForRobot::safe_plan_for_robot(const planning::PlanRequest& request) {
+    Trajectory trajectory;
+    try {
+        trajectory = unsafe_plan_for_robot(request);
+    } catch (std::runtime_error exception) {
```

Comment:
Catch by reference instead of by value according to the github action below.

Reply:
sorry I thought I did this but clearly I did not